### ISSUE TITLE
Postgres `'nan'::decimal` cannot be injected by the cdc backfill

### ISSUE THREAD

**Issue by StrikeW** [risingwavelabs/risingwave#14218]
```
 let v = row.try_get::<_, Option<RustDecimal>>(i)?;
```

Since rust `Decimal` cannot represent `NaN`.

_Originally posted by @xiangjinwu in https://github.com/risingwavelabs/risingwave/pull/13958#discussion_r1427625887_

**Comment by fuyufjh** [comment]
We do have an `NaN` in decimal 

https://github.com/risingwavelabs/risingwave/blob/ae7d7ba426818d43209be545ad72f979ce3dacc0/src/common/src/types/decimal.rs#L44

**Comment by StrikeW** [comment]
> We do have an `NaN` in decimal
> 
> https://github.com/risingwavelabs/risingwave/blob/ae7d7ba426818d43209be545ad72f979ce3dacc0/src/common/src/types/decimal.rs#L44

Yeah, I know that. The problem is that the `rust_decimal` crate cannot represent `NaN`. A upstream `NaN` will be converted to `0`.
